compiler/rustc_hir_typeck/src/callee.rs

```diff
@@ -183,16 +183,23 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     kind: TypeVariableOriginKind::TypeInference,
                     span: callee_expr.span,
                 });
+                // We may actually receive a coroutine back whose kind is different
+                // from the closure that this dispatched from. This is because when
+                // we have no captures, we automatically implement `FnOnce`. This
+                // impl forces the closure kind to `FnOnce` i.e. `i8`.
```

Comment:
uh... explain that `i8` comment please ^^

Reply:
This is the correspondence between Ty and ClosureKind: https://doc.rust-lang.org/stable/nightly-rustc/src/rustc_middle/ty/closure.rs.html#131

but I got it backwards whoops

Reply:
I ended up removing this hack in favor of another one that's a bit easier to understand: 3bb384aad6e7f61a0b4b8c604206e78ffa418df4